lib/ui/popup.js

```diff
@@ -20,7 +27,33 @@ Popup.prototype = {
     this.el = this.win.document.getElementById('PopupAutoCompleteRichResult');
     this.el.addEventListener('popuphiding', this.beforePopupHide);
   },
+  destroy: function() {
+    this.el.removeEventListener('popuphiding', this.beforePopupHide);
+
+    this.win.clearTimeout(this.mouseOverTimeout);
+
+    delete this.el;
+    delete this.win;
+  },
   beforePopupHide: function(evt) {
     this.events.publish('before-popup-hide');
+  },
+  onResultsMouseOver: function(evt) {
+    if (this.ignoreNextMouseOver) {
+      this.ignoreNextMouseOver = false;
+      return;
+    }
+
+    // Throttle mouseover events, which fire rapidly if the user quickly moves
+    // the mouse across different DOM elements in the results list.
+    if (this.mouseOverTimeout) {
+      return;
+    }
+    this.mouseOverTimeout = this.win.setTimeout(() => {
+      this.win.clearTimeout(this.mouseOverTimeout);
+      this.mouseOverTimeout = null;
+    }, 15);
```

Comment:
I use a nearly-identical throttle syntax elsewhere in this PR. I think it's clearer in this form than wrapped in a throttle utility method, but open to alternative points of view :-)